kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,1013 @@
+        self.line['dash_length'] = None
+        self.line['dash_list'] = []
+
+    def set_capstyle(self, cs):
+        '''Set the cap style based on the kivy framework cap styles.
+        '''
+        GraphicsContextBase.set_capstyle(self, cs)
+        self.line['cap_style'] = self._capd[self._capstyle]
+
+    def set_joinstyle(self, js):
+        '''Set the join style based on the kivy framework join styles.
+        '''
+        GraphicsContextBase.set_joinstyle(self, js)
+        self.line['join_style'] = js
+
+    def set_dashes(self, dash_offset, dash_list):
+        GraphicsContextBase.set_dashes(self, dash_offset, dash_list)
+        # dash_list is a list with numbers denoting the number of points
+        # in a dash and if it is on or off.
+        if dash_list is not None:
+            self.line['dash_list'] = dash_list
+        if dash_offset is not None:
+            self.line['dash_offset'] = int(dash_offset)
+
+    def set_linewidth(self, w):
+        GraphicsContextBase.set_linewidth(self, w)
+        self.line['width'] = w
+
+    def _get_style_dict(self, rgbFace):
+        '''Return the style string. style is generated from the
+           GraphicsContext and rgbFace
+        '''
+        attrib = {}
+        forced_alpha = self.get_forced_alpha()
+        if rgbFace is None:
+            attrib['fill'] = 'none'
+        else:
+            if tuple(rgbFace[:3]) != (0, 0, 0):
+                attrib['fill'] = str(rgbFace)
+            if len(rgbFace) == 4 and rgbFace[3] != 1.0 and not forced_alpha:
+                attrib['fill-opacity'] = str(rgbFace[3])
+
+        if forced_alpha and self.get_alpha() != 1.0:
+            attrib['opacity'] = str(self.get_alpha())
+
+        offset, seq = self.get_dashes()
+        if seq is not None:
+            attrib['line-dasharray'] = ','.join(['%f' % val for val in seq])
+            attrib['line-dashoffset'] = six.text_type(float(offset))
+
+        linewidth = self.get_linewidth()
+        if linewidth:
+            rgb = self.get_rgb()
+            attrib['line'] = str(rgb)
+            if not forced_alpha and rgb[3] != 1.0:
+                attrib['line-opacity'] = str(rgb[3])
+            if linewidth != 1.0:
+                attrib['line-width'] = str(linewidth)
+            if self.get_joinstyle() != 'round':
+                attrib['line-linejoin'] = self.get_joinstyle()
+            if self.get_capstyle() != 'butt':
+                attrib['line-linecap'] = _capstyle_d[self.get_capstyle()]
+
+        return attrib
+
+
+def draw_if_interactive():
+    '''Handle whether or not the backend is in interactive mode or not.
+    '''
+    if matplotlib.is_interactive():
+        figManager = Gcf.get_active()
+        if figManager:
+            figManager.canvas.draw_idle()
+
+
+class Show(ShowBase):
+    '''mainloop needs to be overwritten to define the show() behavior for kivy
+       framework.
+    '''
+    def mainloop(self):
+        app = App.get_running_app()
+        if app:
```

Comment:
Ok sorry but I was looking at the code and I need this App to be instantiated with the figurecanvas and the toolbar. I can not see a way I can pass these objects directly to the Show()